tabs/tab_data.py

```
@@ -181,97 +179,138 @@ def _():
+            for col in new_df.columns:
+                unique_vals = new_df[col].dropna().unique()
+                is_numeric = pd.api.types.is_numeric_dtype(new_df[col])
+                if is_numeric and len(unique_vals) > 10:
+                    current_meta[col] = {'type': 'Continuous', 'map': {}, 'label': col}
+                else:
+                    current_meta[col] = {'type': 'Categorical', 'map': {}, 'label': col}
+            
+            return new_df, current_meta
+            
+        except Exception as e:
+            logger.error(f"Error loading file: {e}")
```

Ruff (0.14.10)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)